### PR TITLE
Add a bunch of aliases

### PR DESCRIPTION
**Pull request by irevoire**
Now we can have multiple aliases to the same command.
I also added the alias `x` for `wq` and all the associated variants.

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -2085,269 +2085,267 @@ mod cmd {
...
-            alias: Some("lat"),
+            aliases: &["lat"],
             doc: "Jump to a later point in edit history. Accepts a number of steps or a time span.",
             fun: later,
             completer: None,
         },
         TypableCommand {
             name: "write-quit",
-            alias: Some("wq"),
+            aliases: &["wq", "x"],
```
Comment (archseer) [inline helix-term/src/commands.rs:2158]: Is this also available in vim/kakoune? I was thinking :x could close the underlying buffer instead of the view

Comment (irevoire) [reply helix-term/src/commands.rs:2158]: I was going to tell you yes, but it looks like it's not [*strictly*](https://til.hashrocket.com/posts/2fdb6afb66-difference-between-wq-and-x) the same in vim.
`wq` always write the file to disk while `x` write it only if it has changed.
And in kakoune they are the same I think? I don't find anything talking about it and I don't know where to find it in the code.

Personally, I think these aliases are enough for most people.

**Comment by pickfire** [comment]
I wonder if we really want a duplicate `:x` which do the same thing as `:wq`, do you use `:x`?

File: helix-term/src/commands.rs (hunk 1 above)
Comment (pickfire) [reply helix-term/src/commands.rs:2158]: I can't seemed to find `:x` in kakoune.

Comment (irevoire) [reply helix-term/src/commands.rs:2158]: Oh yes you're right I forgot I had added it to my kakrc!
```
define-command x write-quit
```
There is even an issue about it:
https://github.com/mawww/kakoune/issues/3503

**Comment by irevoire** [comment]
Personally, yes I use it every time; even after 3 weeks of using helix I'm always trying to `:x`.
Then I decided to modify helix on my side and after a week I thought I could create a PR so everyone can use it.

**Comment by raygervais** [comment]
As an individual coming from neovim, I cannot overstate how many times a day I'm using `:x` out of habit / efficiency just to remember it doesn't work. I'd be so happy if this gets added!

File: helix-term/src/commands.rs [hunk 2]
```diff
@@ -2085,269 +2085,267 @@ mod cmd {
...
-        }
-
-        map
+        TYPABLE_COMMAND_LIST
+            .into_iter()
+            .map(|cmd| {
+                std::iter::once((cmd.name, cmd))
+                    .chain(cmd.aliases.into_iter().map(move |&alias| (alias, cmd)))
+            })
+            .flatten()
```
Comment (archseer) [inline helix-term/src/commands.rs:2347]: Could you just use flat_map here?

**Review by archseer — CHANGES_REQUESTED** [review]

File: helix-term/src/commands.rs (hunk 2 above)
Comment (irevoire) [reply helix-term/src/commands.rs]: done :+1:

**Comment by irevoire** [comment]
Is the hacktoberfest thing working with this repo?
I see there is the tag but my contribution was not counted :eyes:

**Comment by archseer** [comment]
> I see there is the tag

Is there? I'm on my phone so I can't see it right now but I was against enabling hacktoberfest.

**Comment by irevoire** [comment]
Yes it's there in the label of the project: 
image

**Comment by sudormrfbin** [comment]
Hacktoberfest was requested in #795 a while ago.

**Comment by pickfire** [comment]
@irevoire is it still being counted? Or otherwise maybe I will try adding `hacktoberfest-accepted` to this PR.

File: helix-term/src/commands.rs [hunk 3]
```diff
@@ -2085,269 +2085,266 @@ mod cmd {
...
-            alias: Some("vs"),
+            aliases: &["vs"],
             doc: "Open the file in a vertical split.",
             fun: vsplit,
             completer: Some(completers::filename),
         },
         TypableCommand {
             name: "hsplit",
-            alias: Some("sp"),
+            aliases: &["hs", "sp"],
```
Comment (pickfire) [inline helix-term/src/commands.rs:2333]: Where does this `hs` come from`?

Comment (irevoire) [reply helix-term/src/commands.rs:2333]: For horizontal-split, it's not used by others editor I think but it looked quite logical to me if we have the `vs`

**Comment by irevoire** [comment]
I don't really know, it's the first time I contribute to a repo with the hacktoberfest enabled.
My pr is marked as "maturing"

**Pull request merged**